listen/range-changes.js

```diff
@@ -1,35 +1,76 @@
 "use strict";
 
-var WeakMap = require("weak-map");
```

Comment:
Eliminating the WeakMap is a good choice. Perhaps eliminating the Dict would be good as well.